ipatests/test_webui/test_host.py

```diff
@@ -407,11 +435,353 @@ def test_indirect_associations(self):
         self.assert_indirect_record(hbac.RULE_PKEY, ENTITY, 'memberof_hbacrule')
         self.assert_indirect_record(sudo.RULE_PKEY, ENTITY, 'memberof_sudorule')
 
-        ## cleanup
-        ## -------
+        # cleanup
+        # -------
         self.delete(ENTITY, [self.data])
         self.delete(hostgroup.ENTITY, [hostgroup.DATA, hostgroup.DATA2])
         self.delete(netgroup.ENTITY, [netgroup.DATA])
         self.delete(rbac.ROLE_ENTITY, [rbac.ROLE_DATA])
         self.delete(hbac.RULE_ENTITY, [hbac.RULE_DATA])
         self.delete(sudo.RULE_ENTITY, [sudo.RULE_DATA])
+
+    @screenshot
+    def test_buttons(self):
+        """Test buttons"""
+        self.init_app()
+        actions = ActionChains(self.driver)
+        self.navigate_to_entity(ENTITY)
+        # add with enter key
+        self.navigate_to_entity(ENTITY)
+        self.button_click('add')
+        self.fill_text('div[name="hostname"] input[name="hostname"]',
+                       self.pkey)
+        self.check_option('force')
+        actions.send_keys(Keys.ENTER).perform()
+        self.wait_for_request(n=3)
+        self.assert_record(self.pkey)
+
+        # add and another
+        self.add_record(ENTITY, [self.data2, self.data3])
+        # webui automatically lower case
+        self.pkey3 = self.pkey3.lower()
+        self.data3['pkey'] = self.data3['pkey'].lower()
+        self.pkey4 = self.pkey4.lower()
+        self.data4['pkey'] = self.data4['pkey'].lower()
+
+        # add and edit record
+        self.add_record(ENTITY, self.data4, dialog_btn='add_and_edit')
+        self.assert_facet(ENTITY, facet="details")
+
+        # cancel managedby
+        self.add_associations([self.pkey3], facet='managedby_host',
+                              confirm_btn="cancel")
+        self.wait()
+        fqdn_table = self.find('table[name="fqdn"]', By.CSS_SELECTOR)
+        self.select_record(self.pkey4, parent=fqdn_table)
```

Comment:
This looks too low-level. Parent serves to provide form or dialog. So here the usage can probably be: `self.select_record(self.pkey4, table_name='fqdn')`